### DISCUSSION TITLE
Separate test files

### DISCUSSION THREAD

**Discussion by brian316** [getgauge/gauge#1865] [General]
How do I implement separate test files to use the POM method (page object model; where each webpage has its own test file) for web automation? I'm using selenium, python, and gauge but I can't find a way to create separate files for each page. I cant use the decorators @after_suite either because if I define some variables in file A and I use steps in file B then the @after_suite in the file will be called as well and result in a fail

**Comment by sriv** [comment]
Do you have an example where your code structure does not work with gauge?

In general, Gauge does not recommend Page Object Model - https://gauge.org/2017/06/13/the-page-objects-anti-pattern/
